subsys/bluetooth/host/l2cap.c

```diff
@@ -1256,6 +1252,25 @@ static void l2cap_chan_seg_sent(struct bt_conn *conn, void *user_data)
 	l2cap_chan_tx_resume(BT_L2CAP_LE_CHAN(chan));
 }
 
+static bool test_and_dec(atomic_t *target)
+{
+	static struct k_spinlock lock;
+	k_spinlock_key_t key;
+
+	key = k_spin_lock(&lock);
```

Comment:
could we use atomic_get and atomic_cas and not have the spin-lock?

Reply:
We could indeed do that, or a get follow by atomic_dec but then it is not an atomic operation but 2 so there is a window where it could be preempted.

Reply:
The window should be handled by having the atomic_cas in a loop until it succeeds AFAIK.

Reply:
I guess we don't want a busy loop though otherwise it is not that different than using a spin_lock.

Reply:
You need to delete this function and do something else, working with the existing atomic operations in Zephyr.